### PR TITLE
Open product screen when onboarding task add product is clicked

### LINKED ISSUE #8628: Open "Add product" flow from onboarding list
Screenshot 2023-03-22 at 16 02 00

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #8628 

### Description
Opens product list screen (which will be empty) when clicking on "Add your first product" task. 

### Testing instructions
- Log into a site that doesn't have any product (create a jurassic ninja site if necessary) 
- Click on "Add your first product" onboarding task
- Check you land empty product list and the bottom sheet to add new product is displayed right away

### Images/gif

https://user-images.githubusercontent.com/2663464/227025211-b553a7af-9c67-42ab-aa31-f2591c090c57.mp4

### CODE REVIEW

File: WooCommerce/src/main/res/navigation/nav_graph_main.xml [hunk 1]
```diff
@@ -40,8 +40,11 @@
             android:id="@+id/action_myStore_to_launchStoreFragment"
             app:destination="@id/launchStoreFragment" />
         <action
-            android:id="@+id/action_dashboard_to_nav_graph_domain_change"
+            android:id="@+id/action_myStore_to_nav_graph_domain_change"
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/res/navigation/nav_graph_main.xml:43]: Just renaming this to match the rest of the actions for the same fragment

**Comment by anitaa1990** [comment]
Hi, sorry for butting in here but I noticed this PR by accident. @JorgeMucientes I noticed that when clicking on the add product button we redirect users to the products tab where they again have to click on the add product button. Can we bypass that and instead just redirect users to the add product flow when they click on the add product task? So not just display the empty product screen but initiate the add product flow automatically.

**Comment by JorgeMucientes** [comment]
Thanks for the feedback @anitaa1990. I entirely agree with your point. I got confused with the design from Figma. I have updated it to show the product type bottom sheet right away when clicking on "Add your first product" task. (See the updated video for reference) 

Please let me know what you think and if we can proceed with merging this :)

**Comment by JorgeMucientes** [comment]
After the most recent discussion: p1679565356325269/1679503846.537109-slack-C045CUK1Y3U
 I'll make one more update to this PR on the behavior. I will post another comment when this is finally ready for review and merging.

**Comment by JorgeMucientes** [comment]
Ready for review and merging @0nko. Sorry for all the constant updated on the behavior. 

The final expected behavior is the following: 

https://user-images.githubusercontent.com/2663464/227190838-26d01b94-3caf-4936-bd6c-be03d465f757.mp4

File: WooCommerce/src/main/res/navigation/nav_graph_main.xml (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/res/navigation/nav_graph_main.xml:43]: 👍

**Review by 0nko — APPROVED** [review]
Looks great! 🏅

**Pull request merged**